### PR TITLE
feat(tests): run unit tests in parallel

### PR DESCRIPTION
**Pull request by khvn26**
Thanks for submitting a PR! Please check the boxes below:

- [x] I have run [`pre-commit`](https://docs.flagsmith.com/deployment/locally-api#pre-commit) to check linting
- [x] I have filled in the "Changes" section below?
- [x] I have filled in the "How did you test this code" section below?

## Changes

Another small-ish QoL update. This PR improves API unit tests duration by adoping [pytest-xdist](https://pytest-xdist.readthedocs.io).

Locally on my M2 I observe the overall duration shrink to ~1 minute from the average of 9 minutes.

Interesting to see how pytest-xdists fares on GitHub runners. To improve execution duration on CI we might want to look into pytest-split instead. However, the main goal is to improve the local run duration.  

From my experience large Django test suites need quite some fixing to be runnable with python-xdist. With Flagsmith we're in luck, however, only one tweak had to be made. Arguably, it did improve correctness too.

Additionally, this PR pins `pip-tools` so newer `pip` versions won't break the package locking (see https://github.com/jazzband/pip-tools/issues/1623).

## How did you test this code?

Ran the tests.

### CODE REVIEW

File: api/requirements-dev.in [hunk 1]
```diff
@@ -6,11 +6,12 @@ pytest-django
 pytest-mock
 django-test-migrations
 black
-pip-tools
+pip-tools>=6.13
```
Comment (matthewelwell) [inline api/requirements-dev.in]: We shouldn't really have to pin anything in the `.in` files. Is there a reason we did this vs just updating in the `requirements-dev.txt` file?

Comment (khvn26) [reply api/requirements-dev.in]: I couldn't get the version bump via simply running `pip-compile` and editing an autogenerated file shouldn't really happen?

Comment (gagantrivedi) [reply api/requirements-dev.in]: No, I think you need to `--upgrade-package`.
e.g: `pip-compile requirements-dev.in --upgrade-package pip-tools` should do the job

Comment (khvn26) [reply api/requirements-dev.in]: Fixed.

**Review by matthewelwell — APPROVED** [review]

**Pull request merged**